codersdk/users.go

```diff
@@ -243,6 +243,18 @@ type LoginWithPasswordResponse struct {
 	SessionToken string `json:"session_token" validate:"required"`
 }
 
+// RequestOneTimePasscodeRequest enables callers to request a one-time-passcode to change their password.
+type RequestOneTimePasscodeRequest struct {
+	Email string `json:"email" validate:"required,email" format:"email"`
+}
+
+// ChangePasswordWithOneTimePasscodeRequest enables callers to change their password when they've forgotten it.
+type ChangePasswordWithOneTimePasscodeRequest struct {
+	Email           string `json:"email" validate:"required,email" format:"email"`
```

Comment:
Thought: The email adds an extra layer of authentication, I suppose, but we could probably drop it in favor of looking up the user via OTP only.

Reply:
We could certainly do that, happy to have a discussion about the trade-offs.

Reply:
Actually, we can make the UX nice even though we require the email here as well as long as the flow is like this:

1. Request OTP to your email (fill email, press submit)
2. Be shown new form to fill OTP (email is pre-filled, hidden or read-only field)

Or, if visiting the password reset page manually, the email can't be pre-filled and is required to be entered manually.

I think that's the best of both worlds (retain security + good UX).

Reply:
Yep, this is the flow I had in my head of how this could work 👍